RSS Reader/RSS Reader/View Controllers/Web Browser/RSSRWebBrowserController.m

```diff
@@ -0,0 +1,212 @@
+    
+    self.webView.scrollView.delegate = self;
+    self.webView.navigationDelegate = self;
+
+    [self startLoading];
+}
+
+
+
+- (void)configureToolbarItems {
+    UIBarButtonItem *safariButton = [[UIBarButtonItem alloc] initWithImage:UIImage.RSSRExplore
+                                                               style:UIBarButtonItemStylePlain
+                                                              target:self
+                                                              action:@selector(openInSafari)];
+    UIBarButtonItem *flexibleSpace = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace
+                                                                                   target:nil
+                                                                                   action:nil];
+    
+    [self setToolbarItems:@[self.backButton, flexibleSpace, self.forwardButton, flexibleSpace, self.reloadButton, flexibleSpace, self.stopButton, flexibleSpace, safariButton]];
+    
+    [safariButton release];
+    [flexibleSpace release];
+}
+
+- (void)configureNavigationController {
+    [self.navigationController setHidesBarsOnSwipe:NO];
+    [self.navigationController setToolbarHidden:NO];
+    self.navigationItem.title = [NSURL URLWithString:self.request.URL.absoluteString].host;
+    
+    UIBarButtonItem *doneBackButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone
+                                                                                    target:self
+                                                                                    action:@selector(popViewController)];
+    self.navigationItem.leftBarButtonItem = doneBackButton;
+    [doneBackButton release];
+}
+
+- (void)popViewController {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (void)startLoading {
+    [self.webView loadRequest:self.request];
+}
+
+- (void)disableControlButtons {
+    self.backButton.enabled = NO;
+    self.forwardButton.enabled = NO;
+    self.stopButton.enabled = NO;
+    self.reloadButton.enabled = NO;
+}
+
+
+#pragma mark - Hide/show toolbar and navigationbar
+
+- (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView {
+    self.lastContentOffset = scrollView.contentOffset.y;
+}
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    if (scrollView.contentOffset.y <= self.lastContentOffset) {
+        [self.navigationController setToolbarHidden:NO animated:YES];
+        [self.navigationController setNavigationBarHidden:NO animated:YES];
+    } else {
+        [self.navigationController setToolbarHidden:YES animated:YES];
+        [self.navigationController setNavigationBarHidden:YES animated:YES];
+    }
+}
+
+
+#pragma mark - WKNavigationDelegate
+
+- (void)webView:(WKWebView *)webView didCommitNavigation:(WKNavigation *)navigation {
+    self.backButton.enabled = [self.webView canGoBack] ? YES : NO;
```

Comment:
Just like this:
self.backButton.enabled = [self.webView canGoBack];
self.forwardButton.enabled = [self.webView canGoForward];

Reply:
Done